ni_measurementlink_service/_internal/parameter/metadata.py

```diff
@@ -53,17 +58,76 @@ def validate_default_value_type(parameter_metadata: ParameterMetadata) -> None:
     expected_type = type(
         Context.get_type_default(parameter_metadata.type, parameter_metadata.repeated)
     )
-
-    if not isinstance(default_value, expected_type):
-        raise TypeError(
-            f"Unexpected type {type(default_value)} in the default value for '{display_name}'. Expected type: {expected_type}."
-        )
+    has_enum_values_annotation, enum_values_annotation = try_get_enum_values_annotation(
+        parameter_metadata
+    )
 
     if parameter_metadata.repeated:
+        if not isinstance(default_value, expected_type):
+            raise TypeError(
+                f"Unexpected type {type(default_value)} in the default value for '{display_name}'. Expected type: {expected_type}."
+            )
         expected_element_type = type(Context.get_type_default(parameter_metadata.type, False))
-        for element in default_value:
-            if not isinstance(element, expected_element_type):
+        if has_enum_values_annotation:
+            for element in default_value:
+                if not _is_default_enum_match_annotations(element, enum_values_annotation):
+                    raise TypeError(
+                        f"The default value, `{element}`, for '{display_name}' is not valid. Expected values: `{enum_values_annotation}`."
```

Comment:
IMO this buries the lede. "Invalid default value X for enum Y"?

Reply:
Rewored as suggested.